src/main/java/org/folio/oaipmh/dao/impl/ConfigurationSettingsDaoImpl.java

```diff
@@ -0,0 +1,201 @@
+package org.folio.oaipmh.dao.impl;
+
+import static org.folio.rest.jooq.Tables.CONFIGURATION_SETTINGS;
+
+import io.github.jklingsporn.vertx.jooq.classic.reactivepg.ReactiveClassicGenericQueryExecutor;
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.pgclient.PgException;
+import java.util.Map;
+import java.util.UUID;
+import javax.ws.rs.NotFoundException;
+import org.folio.oaipmh.dao.ConfigurationSettingsDao;
+import org.folio.oaipmh.dao.PostgresClientFactory;
+import org.folio.oaipmh.exception.ConfigSettingException;
+import org.jooq.impl.DSL;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class ConfigurationSettingsDaoImpl implements ConfigurationSettingsDao {
+
+  private static final String ALREADY_EXISTS_ERROR_MSG =
+        "Configuration " + "setting with id '%s' already exists";
+  private static final String NOT_FOUND_ERROR_MSG =
+        "Configuration setting with id '%s' was not found";
+  private static final String CONFIG_NAME_NOT_FOUND_ERROR_MSG =
+        "Configuration setting with config name '%s' was not found";
+
+  private final PostgresClientFactory postgresClientFactory;
+
+  public ConfigurationSettingsDaoImpl(final PostgresClientFactory postgresClientFactory) {
+    this.postgresClientFactory = postgresClientFactory;
+  }
+
+  @Override
+  public Future<JsonObject> getConfigurationSettingsById(String id, String tenantId) {
+    return getQueryExecutorReader(tenantId).transaction(txQE ->
+      txQE.findOneRow(dslContext ->
+          dslContext.selectFrom(CONFIGURATION_SETTINGS)
```

Comment:
mb there are issues with codestyle, in some places you have 2 spaces indent, in some 4